### DISCUSSION TITLE
Full-height div with AppShell?

### DISCUSSION THREAD

**Discussion by davidchalifoux** [mantinedev/mantine#5272] [Q&A] [answered]
I'm struggling to get a full-height div as a child to `<AppShell.Main>`. Setting `height: 100%` doesn't work. I found [this previous discussion](https://github.com/orgs/mantinedev/discussions/3588), but I don't think it's an ideal solution. Any tips?

Here's a [CodeSandbox ](https://codesandbox.io/p/sandbox/agitated-engelbart-7k6y3r?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clp21uvw50006356gb981zla8%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clp21uvw50003356gz6j3de1i%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clp21uvw50004356gi4d3mjsf%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clp21uvw50005356gtx5ifl60%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B60.06912268196006%252C39.93087731803994%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clp21uvw50003356gz6j3de1i%2522%253A%257B%2522id%2522%253A%2522clp21uvw50003356gz6j3de1i%2522%252C%2522activeTabId%2522%253A%2522clp22ni5400q2356guoo7nk0i%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clp21uvw50002356gcxq64yjh%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252FREADME.md%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252F.devcontainer%252Fdevcontainer.json%2522%252C%2522id%2522%253A%2522clp21yd5c002u356grvieeimo%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252F.codesandbox%252Ftasks.json%2522%252C%2522id%2522%253A%2522clp221jid00i1356g6v6p7kk3%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fapp%252Fpage.tsx%2522%252C%2522id%2522%253A%2522clp22ni5400q2356guoo7nk0i%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%257D%252C%2522clp21uvw50005356gtx5ifl60%2522%253A%257B%2522id%2522%253A%2522clp21uvw50005356gtx5ifl60%2522%252C%2522activeTabId%2522%253A%2522clp221a1600gf356gg7r1v2wi%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522DOCS%2522%252C%2522id%2522%253A%2522clp21vjjw006z356gs2tngek2%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%252C%257B%2522type%2522%253A%2522SANDBOX_INFO%2522%252C%2522id%2522%253A%2522clp21wtcw00aq356gjdd60kaa%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%252C%257B%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A3000%252C%2522id%2522%253A%2522clp221a1600gf356gg7r1v2wi%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522path%2522%253A%2522%252F%2522%257D%255D%257D%252C%2522clp21uvw50004356gi4d3mjsf%2522%253A%257B%2522id%2522%253A%2522clp21uvw50004356gi4d3mjsf%2522%252C%2522activeTabId%2522%253A%2522clp2214s900eb356g2me2co9g%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522id%2522%253A%2522clp2214s900eb356g2me2co9g%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Afalse%252C%2522sidebarPanelSize%2522%253A15%257D)example.

**Comment by xyzCoding34** [comment]
Hows your code block? use a main component as a child into appshell with mantines <Grid/>.
you can check useViewPortSize hook btw...

**Reply by davidchalifoux** [reply]
Check the CodeSandbox for a minimal example.
I don't think I should have to use a hook to make a div full-height.

**Comment by davidchalifoux** [comment, marked as answer]
A user on Discord gave a good solution (Thanks Cyan):
```
<AppShell.Main style={{ height: "100dvh" }}>
  <div style={{ height: "100%" }}>
    This should full height
  </div>
</AppShell.Main>
```
